### PR TITLE
Yewscion changes

### PR DESCRIPTION
**Pull request by yewscion**
Hello All,

I've made a number of changes to the way the code works here, but I believe they are for the better.

I'll mostly let the commits speak for themselves, but in short, this PR:

- Changes the default interpreter to "bqn" instead of "BQN".
- Fixes the BQN Keymap to assume a font is used to display the bqn glyphs.
- Adds a secondary helper buffer, bqn-glyphs-mode, that is a table with symbol names.
- Fixes the bqn-process-execute-region procedure to properly handle multiline listings for CBQN, per https://github.com/dzaima/CBQN/issues/29.
- Adds a bqn-process-execute-region-follow procedure.
- Adds default keybinds for the execute procedures.
- Shortens the displayed names of some of the more verbose glyphs so the glyphs-mode table can fit in < 80 column displays.

### CODE REVIEW

**Comment by yewscion** [comment]
Opened Issue #8 to track this PR as an issue.

**Comment by yewscion** [comment]
Please let me know if there are any concerns found within these commits; I am more than willing to help address them in order to get these changes merged.

**Comment by AndersonTorres** [comment]
Sorry by the delay. I will look at them along this week.

File: bqn-comint.el [hunk 1]
```diff
@@ -118,7 +118,9 @@ When DONT-FOLLOW is non-nil, maintain focus on the buffer where the function was
         (buffer (current-buffer)))
     (pop-to-buffer (process-buffer session))
     (goto-char (point-max))
-    (insert (format "\n%s\n" region))
+    (insert (format ")escaped \"%s\""
```
Comment (AndersonTorres) [inline bqn-comint.el]: Reword this commit as below:

> Enables buffer and region evals.

> Following the recommendations from https://github.com/dzaima/CBQN/issues/29

As a rule of thumb, the title should be a short, imperative description of the change you want to implement. The body of commit message is filled with the comments you deem necessary.

File: bqn-glyphs.el [hunk 2]
```diff
@@ -71,3 +71,4 @@
 	  (bqn-glyph-mode))
         (set-window-buffer window buffer)
         (fit-window-to-buffer window)))))
+(provide 'bqn-glyphs)
```
Comment (AndersonTorres) [inline bqn-glyphs.el:74]: New files should be added in a single commit. Squash those smaller mistakes.

Comment (AndersonTorres) [reply bqn-glyphs.el:74]: Being more accurate: if you are adding a new file, it should not be split between commits.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by yewscion** [comment]
I have reworded and squashed my changes into two commits: One solely for the keyboard/glyphs/fonts/etc, and one for the region/buffer execution and configuration. Let me know if this is acceptable!

File: bqn-comint.el [hunk 3]
```diff
@@ -101,28 +79,42 @@ If it doesn't exist, create and return it; else, return the existing one."
       (switch-to-buffer-other-window *bqn-process-buffer-name*)
       (bqn-inferior-mode)
       (set-input-method "BQN-Z"))))
-
-(defun bqn-process-execute-region (start end &optional dont-follow)
-  "Send the current region to the bqn-process-session.
-
-When DONT-FOLLOW is non-nil, maintain focus on the buffer where the function was called from."
+(defun bqn-process-execute-region (start end &optional dont-follow return-output)
```
Comment (AndersonTorres) [inline bqn-comint.el:82]: 1. Add a blank line between functions;
2. Docstring them:
   
   https://www.gnu.org/software/emacs/manual/html_node/elisp/Documentation-Tips.html

3. More generally, [follow the guide](https://www.gnu.org/software/emacs/manual/html_node/elisp/Coding-Conventions.html).

File: bqn-comint.el [hunk 4]
```diff
@@ -101,28 +79,42 @@ If it doesn't exist, create and return it; else, return the existing one."
...
-    (bqn--flash-region start end))
   (let ((region (buffer-substring-no-properties start end))
         (session (bqn-process-ensure-session))
         (buffer (current-buffer)))
-    (pop-to-buffer (process-buffer session))
-    (goto-char (point-max))
-    (insert (format "\n%s\n" region))
+    (bqn-process-execute-region-setup start end session region)
+    (bqn-process-execute-region-execute buffer)))
+(defun bqn-process-execute-region-setup (start end session region)
```
Comment (AndersonTorres) [inline bqn-comint.el:89]: If the function should be hidden, not meant to be consumed by the end user, prefix it with double-hyphen, `bqn--`

https://github.com/bbatsov/emacs-lisp-style-guide#naming

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
I have merged some changes, and the other I tried to understand without success.

**Comment by yewscion** [comment]
Copy that. I apologize for my radio silence, other things have been demanding my attention.

I will review what was merged and fix what wasn't. Thank You for Your efforts and communication.

**Comment by AndersonTorres** [comment]
Closing as unable to understand and full of conflicting entries.

Feel free to reopen.

**Pull request closed without merging**